packages/rest/src/body-parsers/body-parser.text.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {RequestBodyParserOptions} from '../types';
+
+import {text} from 'body-parser';
+import {inject} from '@loopback/context';
+import {Request} from '../..';
+import {RestBindings} from '../keys';
+import {is} from 'type-is';
+
+import {RequestBody, BodyParser} from './types';
+import {
+  BodyParserWithCallback,
+  DEFAULT_LIMIT,
+  getParserOptions,
+  parseRequest,
+} from './helper';
+
+export class TextBodyParser implements BodyParser {
+  name = 'text';
+  private textParser: BodyParserWithCallback;
+
+  constructor(
+    @inject(RestBindings.REQUEST_BODY_PARSER_OPTIONS, {optional: true})
+    options: RequestBodyParserOptions = {},
+  ) {
+    const textOptions = Object.assign(
+      {type: 'text/*', limit: DEFAULT_LIMIT},
+      getParserOptions('text', options),
+    );
+    this.textParser = text(textOptions);
+  }
+
+  supports(mediaType: string) {
+    return !!is(mediaType, 'text/*');
```

Comment:
Similarly here - why is JsonParser using `json` but TextParser `text/*`?

Reply:
> If you decide to keep `text/*`, then please add a test showing what happens when the user registers a parser that accepts a more specific text-based type, e.g. `text/json` or `text/html`.

This is basically what I repeated in my review comment later, where I proposed `text/json`.

Reply:
I extended json parser to deal with `*/json` and `*/*+json`.

Reply:
> I extended json parser to deal with */json and */*+json.

I am afraid you missed my point.

I was using `text/json` as an example use case (test case), where the user wants to register a specialized parser for a media type that's already covered by a built-in parser. 

I see that the current implementation sorts the parsers and invokes built-in parsers at the end, so I suppose this is solved (although I didn't check if there is an accompanying test).